source/exe/main.cc

```diff
@@ -33,10 +35,32 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+int validateConfigAndExit(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
```

Comment:
This would be better placed in an anonymous namespace to make it local to the translation unit (similar to a static function, but the C++ way is to use namespaces).